docs/migrate-to-antrea.md

```diff
@@ -0,0 +1,63 @@
+# Migrate from other CNI to Antrea
+
+This document describes how to migrate from other CNIs to Antrea.
+
+NOTE: The following is a reference list of CNIs and versions we have already
+verified the migration process. CNIs and versions that are not listed here
+might also work, please create an issue if you run into problems during the
+migration to Antrea. In addition, during the migration process, no Kubernetes
+resources should be created or deleted, otherwise the migration process might
+fail or some unexpected problems might occur.
+
+| CNI     | Version |
+|---------|---------|
+| Calico  | v3.26   |
+| Flannel | v0.22.0 |
+
+The migration process is divided into three steps:
+
+1. Clean up old CNI.
+2. Install Antrea in the cluster.
+3. Restart all Pods in the cluster in-place.
+
+## Clean up old CNI
+
+Before installing Antrea in the cluster, the old CNI needs to be cleaned up. The
+cleanup process is different for different CNIs, but the basic idea is to
+remove the old CNI DaemonSet, binaries and configuration files from the cluster.
+If the old CNI use iptables to manage the network, the iptables rules need to be
+cleaned up as well.
```

Comment:
It should be clear which steps should be done by users and which are by processes. My point is we should ask users to do the easy and method-rich part themselves, e.g. the uninstallation of the explicit assets like DaemonSet, Deployment, CRD, and we do the implicit part automatically with best efforts, and can increase the cleanup step by step.
The reasons were because:
1. There are too many installation methods and versions of CNIs need to customize once the migrator is made specific to methods and versions, while a CNI should work in a consistent way despite of the installation method.
2. Uninstalling a CNI's explicit assets like DS, Deployment, and CRD is not a hard work for users but would be a bit complicated for a migrator, including the CNI discovery, installation method discovery, permission issue (we should avoid acquiring delete permission of DS, Deployment, CRD)
3. Uninstalling a CNI's datapath is a CNI specific job but there is 1 case need to handle anyway, we can do our best to clean them up and some stale resources may not matter if they are not deleted.

For the doc, I think we should be clear that users need to uninstall the old CNI themselves, to remove the old CNI's DaemonSet, Deployment, CRD, etc.  And for those binaries, configuration files, iptables rules, they will be taken care by the migrator, or they don't need to be touched (like the binary), so don't document it like users need to do it themselves, otherwise it lookes quite complicated to run the migrator.